flink/src/test/java/org/apache/iceberg/flink/SimpleDataUtil.java

```diff
@@ -129,11 +129,25 @@ public static void assertTableRows(String tablePath, List<RowData> expected) thr
   public static void assertTableRecords(Table table, List<Record> expected) throws IOException {
     table.refresh();
     try (CloseableIterable<Record> iterable = IcebergGenerics.read(table).build()) {
-      Assert.assertEquals("Should produce the expected record",
-          Sets.newHashSet(expected), Sets.newHashSet(iterable));
+      List<Record> iterList = Lists.newArrayList(iterable.iterator());
+      sortRecordList(iterList);
+      sortRecordList(expected);
+      Assert.assertEquals("Should produce the expected record", expected, iterList);
     }
   }
 
+  private static void sortRecordList(List<Record> list) {
+    Collections.sort(list, (o1, o2) -> {
```

Comment:
Using hashCode to sort this list is correct if we only want to check wether the two list is equal or not.  But it may not friend to debug the root cause when two list is not equal because those elements are disordered. I'd prefer to compare the fields in lexicographical order if possible.

Reply:
will there be some unknown data type in the Record that does not implement the compareTo interface?

Reply:
Yeah, if this is just for comparing two lists, then it would be better to use a set.

Reply:
The assertTableRecords method provides a `List` parameter. For a new user  to  use this method, he may not know to use `set` when comparing data.

Whether there is such a situation: the original data is (1,'a'),(1,'a'), but the actual running result is (1,'a'). This test case passed, but it is wrong.

Reply:
Maybe we should remove the duplicate records from tests so that we can use a set?

Reply:
We can generate non-repetitive test data when writing test cases, but there must be repeated data in the production environment. Will there be test cases that pass but the production environment is incorrect?

Reply:
In this SimpleDataUtil, all `Record` should have the same schema (`id` and `data`  column), so  I think we could just compare those two fields directly.  Don't have to compare the hashCode.

About using `set` to compare lists,  I'd prefer to compare sorted lists.  Because in future equality-delete test cases,  we may delete the same row several times to make sure that we've gurantteed the correct semantics.

Reply:
In order to test the multi-level partition filter, I constructed a 2-level partition in this method : [TestRewriteDataFilesAction#testRewriteDataFilesWithFilter](https://github.com/apache/iceberg/blob/master/flink/src/test/java/org/apache/iceberg/flink/actions/TestRewriteDataFilesAction.java#L190) , so each record has three fields. I don’t know if there will be similar test cases in the future.

Reply:
Well,  how about using the `com.google.common.collect.Multiset`  to assert those lists with duplicated records  ? Don't have to sort based on hashCode,  and it also allow to use more fields.

Reply:
Yes,it is great ,I update the pr.